src/reflection.ts

```diff
@@ -463,6 +464,12 @@ export class Parser {
     return (t: Table) => {
       const offset = t.offset + t.bb.__offset(t.offset, field.offset());
       if (offset === t.offset) {
+        // If the field is marked as 'optional' in the schema (i.e., its default is `null`),
+        // return `null` when the field is absent. This preserves FlatBuffers' optional-scalar semantics.
+        if (typeof field.optional === "function" && field.optional()) {
```

Comment:
When would `typeof field.optional === "function"` be false?

Should we consider hoisting the optional() call out of this closure since it does not change? (although, it matches how we do offset() which also doesn't change)

Reply:
Also curious about when the `typeof` is relevant.

As for hoisting it out, I'd be in favor of it, but lack a strong opinion. The cost of this call is going to be a bit higher than the `field.offset()` call.